strings/base_coroutine_threadpool.h

```diff
@@ -370,17 +390,19 @@ WINRT_EXPORT namespace winrt
             return false;
         }
 
-        void await_resume() const noexcept
+        void await_resume() const
         {
+            failure.check();
         }
 
-        void await_suspend(impl::coroutine_handle<> handle) const
+        void await_suspend(impl::coroutine_handle<> handle)
```

Comment:
Nope, can't do this. It prevents you from `co_await`'ing a `const apartment_context`. I'll have to give the `apartment_context` an `operator co_await`.